staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -287,11 +290,16 @@ func (a *Authenticator) AuthenticateToken(token string) (user.Info, bool, error)
 		// Check the email_verified claim to ensure the email is valid.
 		// https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
 		var emailVerified bool
-		if err := c.unmarshalClaim("email_verified", &emailVerified); err != nil {
-			return nil, false, fmt.Errorf("oidc: parse 'email_verified' claim: %v", err)
+		err := c.unmarshalClaim("email_verified", &emailVerified)
+		if err == nil {
+			// If the email_verified claim is present we have to verify it is set to `true`.
+			if !emailVerified {
+				return nil, false, fmt.Errorf("oidc: email not verified")
+			}
 		}
-		if !emailVerified {
-			return nil, false, fmt.Errorf("oidc: email not verified")
+		// If the email_verified claim is not present we do not have to verify it.
+		if err != nil && err != errClaimNotFound {
```

Comment:
it seems like it would be clearer to check for presence of the claim before calling unmarshalClaim, rather than comparing literal errors

```go
if _, hasEmailVerifiedClaim := c["email_verified"]; hasEmailVerifiedClaim {
  ...
}
```